### ISSUE TITLE
SSR Support

### ISSUE THREAD

**Issue by ixxie** [robinscholz/svelte-inline-svg#6]
When using `svelte-inline-svg` I get the error:

```
Error: <InlineSVG> is not a valid SSR component. You may need to review your build config to ensure that dependencies are compiled, rather than imported as pre-compiled modules
    at validate_component (/home/ixxie/repos/coflow/node_modules/svelte/internal/index.mjs:1343:18)
    at /home/ixxie/repos/coflow/__sapper__/dev/server/server.js:343:50
    at Object.$$render (/home/ixxie/repos/coflow/node_modules/svelte/internal/index.mjs:1366:23)
    at /home/ixxie/repos/coflow/__sapper__/dev/server/server.js:362:157
    at Object.$$render (/home/ixxie/repos/coflow/node_modules/svelte/internal/index.mjs:1366:23)
    at Object.default (/home/ixxie/repos/coflow/__sapper__/dev/server/server.js:373:64)
    at /home/ixxie/repos/coflow/__sapper__/dev/server/server.js:312:83
    at Object.$$render (/home/ixxie/repos/coflow/node_modules/svelte/internal/index.mjs:1366:23)
    at /home/ixxie/repos/coflow/__sapper__/dev/server/server.js:372:36
    at Object.$$render (/home/ixxie/repos/coflow/node_modules/svelte/internal/index.mjs:1366:23)
 ```
 
 In my log I also get informed:
 
 ```
 [rollup-plugin-svelte] The following packages did not export their `package.json` file so we could not check the "svelte" field. If you had difficulties importing svelte components from a package, then please contact the author and ask them to export the package.json file.

- svelte-inline-svg
```

Am I doing something wrong or is this indeed as the error says, an issue with exports in this package?

**Comment by robinscholz** [comment]
@ixxie Could you provide a repository with a minimal test case? This might be due to recent changes with Svelte, will have to look into it!

**Comment by ixxie** [comment]
No problem, [here you go](https://github.com/ixxie/InlineSVG-SSR-Issue).

**Comment by robinscholz** [comment]
@ixxie If you install the package as a development dependency. See https://github.com/sveltejs/sapper-template#using-external-components for more info. I'll add a disclaimer to the readme.

**Comment by ixxie** [comment]
This didn't work for me... I pushed [a commit of a minimal example with the package installed as a development dependency](https://github.com/ixxie/InlineSVG-SSR-Issue).

**Comment by ixxie** [comment]
The link you mentioned mentions that you can also "modify the external option under server in rollup.config.js or the externals option in webpack.config.js" - the problem is it doesn't say _how_ to modify it....

**Comment by robinscholz** [comment]
Your `index.svelte` file links to a non-existing SVG.

```
<script>
    import InlineSVG from 'svelte-inline-svg'
</script>

<svelte:head>
	<title>InlineSVG SSR Issue</title>
</svelte:head>

<h1>Great failure!</h1>

<figure>
	<InlineSVG src='../test.svg'/> // Remove static from the path
	<!-- <figcaption>Behold - error!</figcaption> -->
</figure>
```

**Comment by ixxie** [comment]
The [SVG does exist](https://github.com/ixxie/InlineSVG-SSR-Issue/blob/master/static/test.svg) but it's true that path I set was wrong; but I didn't have luck with the path you propose either (which makes sense because. I also tried to set `src='../../static/test.svg'` and to move the SVG to the same folder as `index.svelte` but that didn't work either.

The error message leads me to believe the issue is not related to the source path, but to SSR issues:

![Screenshot from 2021-02-14 15-56-20](https://user-images.githubusercontent.com/20320695/107878775-784be680-6edd-11eb-97e0-037f3b0e989a.png)

**Comment by robinscholz** [comment]
You need to include the package as a dev dependency and remove `static` from your import path. It works without a problem once I make both of these changes.

**Comment by ixxie** [comment]
Sorry but [I'm trying this](https://github.com/ixxie/InlineSVG-SSR-Issue/blob/master/src/routes/index.svelte) but somehow still getting the same error :/  what did I do wrong?

**Comment by kanh48** [comment]
You should check [this one](https://sapper.svelte.dev/docs#Third-party_libraries_that_depend_on_window) from Sapper docs

**Comment by robinscholz** [comment]
I was finally able to reproduce your issue using another computer. Will have to look into it once I have some more time. Probably a trivial fix.

<img width="1680" alt="Screenshot 2021-02-16 at 15 12 05" src="https://user-images.githubusercontent.com/8195463/108074538-9cd4c980-7069-11eb-9108-8afc4f43e8cf.png">
<img width="1680" alt="Screenshot 2021-02-16 at 15 12 31" src="https://user-images.githubusercontent.com/8195463/108074570-a78f5e80-7069-11eb-9621-b4b7f26769b0.png">

**Comment by robinscholz** [comment]
Fixed in 1.0.6 by adding a self referencing export to package.json (See: https://nodejs.org/api/packages.html#packages_exports)

**Comment by ixxie** [comment]
Thanks @robinscholz !
